runtime/stdlib/random.go

```diff
@@ -26,26 +26,64 @@ import (
 	"github.com/onflow/cadence/runtime/sema"
 )
 
-const unsafeRandomFunctionDocString = `
+const revertibleRandomFunctionDocString = `
 Returns a pseudo-random number.
 
 NOTE: The use of this function is unsafe if not used correctly.
 
 Follow best practices to prevent security issues when using this function
 `
 
-var unsafeRandomFunctionType = &sema.FunctionType{
+var revertibleRandomFunctionType = &sema.FunctionType{
 	ReturnTypeAnnotation: sema.NewTypeAnnotation(
 		sema.UInt64Type,
 	),
 }
 
-type UnsafeRandomGenerator interface {
+type RandomGenerator interface {
 	// ReadRandom reads pseudo-random bytes into the input slice, using distributed randomness.
 	ReadRandom([]byte) error
 }
 
-func NewUnsafeRandomFunction(generator UnsafeRandomGenerator) StandardLibraryValue {
+func NewRevertibleRandomFunction(generator RandomGenerator) StandardLibraryValue {
+	return NewStandardLibraryFunction(
+		"revertibleRandom",
+		revertibleRandomFunctionType,
+		revertibleRandomFunctionDocString,
+		func(invocation interpreter.Invocation) interpreter.Value {
+			return interpreter.NewUInt64Value(
+				invocation.Interpreter,
+				func() uint64 {
+					var buffer [8]byte
+					var err error
+					errors.WrapPanic(func() {
+						err = generator.ReadRandom(buffer[:])
+					})
+					if err != nil {
+						panic(interpreter.WrappedExternalError(err))
+					}
+					return binary.LittleEndian.Uint64(buffer[:])
+				},
+			)
+		},
+	)
+}
+
+// `unsafeRandom` related constants and functions will be deleted
+// when the function is deprecated
+const unsafeRandomFunctionDocString = `
+Warning: this function will be deprecated, use revertibleRandom function instead. 
```

Comment:
Use "deprecated" instead, as it is detected by the tooling (https://github.com/onflow/cadence-tools/blob/9f82dd687af84e8b6d2ec21c52d80a2b42146009/lint/deprecated_member_analyzer.go#L58) and annotates uses of the function with a warning:

```suggestion
Deprecated: Use revertibleRandom instead.
```